archaius-core/src/main/java/com/netflix/archaius/config/AbstractConfig.java

```diff
@@ -201,130 +201,130 @@ public Short getShort(String key) {
             return Short.parseShort(value);
         }
         catch (NumberFormatException e) {
-            return parseError(e);
+            return parseError(key, value, e);
         }
     }
 
     @Override
     public Short getShort(String key, Short defaultValue) {
-        String value = getString(key);
+        String value = getRawString(key);
         if (value == null) 
             return notFound(defaultValue);
         try {
             return Short.parseShort(value);
         }
         catch (NumberFormatException e) {
-            return parseError(e, defaultValue);
+            return parseError(key, value, e);
         }
     }
 
     @Override
     public BigInteger getBigInteger(String key) {
-        String value = getString(key);
+        String value = getRawString(key);
         if (value == null) 
             return notFound();
         try {
             return BigInteger.valueOf(Long.valueOf(value));
```

Comment:
Not related to the specific change in this PR, but shouldn't this do something like:

```java
return new BigInteger(value);
```

So the value doesn't have to fit within a Long?